Modules/ITS/src/ITSClusterCheck.cxx

```diff
@@ -33,50 +33,40 @@ void ITSClusterCheck::configure() {}
       auto* h = dynamic_cast<TH2D*>(iter->second->getObject());
-
-      if (iter->second->getName().find("Layer0") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 0);
-      if (iter->second->getName().find("Layer1") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 1);
-      if (iter->second->getName().find("Layer2") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 2);
-      if (iter->second->getName().find("Layer3") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 3);
-      if (iter->second->getName().find("Layer4") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 4);
-      if (iter->second->getName().find("Layer5") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 5);
-      if (iter->second->getName().find("Layer6") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 6);
+      for (int ilayer = 0; ilayer < NLayer; ilayer++) {
+        result.addMetadata(Form("Layer%d", ilayer), "good");
+        if (iter->second->getName().find(Form("Layer%d", ilayer)) != std::string::npos && h->GetMaximum() > averageClusterSizeLimit[ilayer]) {
+          result.updateMetadata(Form("Layer%d", ilayer), "bad");
+          result.set(Quality::Bad);
+        }
+      }
     }
 
     if (iter->second->getName().find("ClusterOccupation") != std::string::npos) {
       auto* h = dynamic_cast<TH2D*>(iter->second->getObject());
 
-      if (iter->second->getName().find("Layer0") != std::string::npos && h->GetMaximum() > 40)
-        result = result.getLevel() | (1 << 7);
-      if (iter->second->getName().find("Layer1") != std::string::npos && h->GetMaximum() > 30)
-        result = result.getLevel() | (1 << 8);
-      if (iter->second->getName().find("Layer2") != std::string::npos && h->GetMaximum() > 20)
-        result = result.getLevel() | (1 << 9);
-      if (iter->second->getName().find("Layer3") != std::string::npos && h->GetMaximum() > 30)
-        result = result.getLevel() | (1 << 10);
-      if (iter->second->getName().find("Layer4") != std::string::npos && h->GetMaximum() > 25)
-        result = result.getLevel() | (1 << 11);
-      if (iter->second->getName().find("Layer5") != std::string::npos && h->GetMaximum() > 15)
-        result = result.getLevel() | (1 << 12);
-      if (iter->second->getName().find("Layer6") != std::string::npos && h->GetMaximum() > 14)
-        result = result.getLevel() | (1 << 13);
+      for (int ilayer = 0; ilayer < NLayer; ilayer++) {
+        result.addMetadata(Form("Layer%d", ilayer), "good");
+        if (iter->second->getName().find(Form("Layer%d", ilayer)) != std::string::npos && h->GetMaximum() > clusterOccupationLimit[ilayer]) {
+          result.updateMetadata(Form("Layer%d", ilayer), "bad");
```

Comment:
Let's put quality::medium (the same in the `beautify` function) here because depending on the colliding system, these numbers change so we can put "Medium --> (notify expert, do not call)" as a final message.

Reply:
Can you apply the checks on the TH2Poly `General_Occupancy` instead of the TH2 `ClusterOccupation`? The TH2Poly is the one used by shifters so now we can transfer the checks on that plot.